File: storydoc-backend/src/main/java/io/storydoc/blueprint/element/CompositeBluePrintElement.java
package io.storydoc.blueprint.element;

import io.storydoc.blueprint.classification.Classification;
import io.storydoc.blueprint.layout.FlexLayout;
import io.storydoc.blueprint.layout.Layout;
import lombok.Getter;

import java.util.List;

@Getter
abstract public class CompositeBluePrintElement extends BluePrintElement {

    public CompositeBluePrintElement(String type, String name, Layout layout, List<BluePrintElement> subElements, Classification classifier) {
        super(type, name, classifier);
        this.layout = layout;
        this.subElements = subElements;
    }

    private Layout layout = new FlexLayout();

    private List<BluePrintElement> subElements;

}

File: storydoc-frontend/src/app/api/models/composite-blue-print-element.ts
/* tslint:disable */
/* eslint-disable */
import { BluePrintElement } from './blue-print-element';
import { Classification } from './classification';
import { Layout } from './layout';
export interface CompositeBluePrintElement extends BluePrintElement {
  classification?: Classification;
  layout?: Layout;
  name?: string;
  subElements?: Array<BluePrintElement>;
  type?: string;
}
